pkg/preparer/orchestrate.go

```diff
@@ -84,22 +89,34 @@ func (p *Preparer) WatchForPodManifestsForNode(quitAndAck chan struct{}) {
 		case err := <-errChan:
 			p.Logger.WithError(err).
 				Errorln("there was an error reading the manifest")
-		case results := <-podChan:
-			for _, result := range results {
-				podId := result.Manifest.ID()
-				if podChanMap[podId] == nil {
-					// No goroutine is servicing this app currently, let's start one
-					podChanMap[podId] = make(chan pods.Manifest)
-					quitChanMap[podId] = make(chan struct{})
-					go p.handlePods(podChanMap[podId], quitChanMap[podId])
+		case intentResults := <-podChan:
+			realityResults, _, err := p.store.ListPods(kp.RealityPath(p.node))
+			if err != nil {
+				errChan <- err
+			} else {
+				// if the preparer's own ID is missing from the intent set, we
+				// assume it was damaged and discard it
+				if !checkResultsForID(intentResults, POD_ID) {
+					p.Logger.NoFields().Errorln("Intent results set did not contain p2-preparer pod ID")
```

Comment:
Lets add a bit more color to this message - "... pod ID - it is possible that Consul's data is corrupt". This might help someone looking at preparer logs who isn't aware of this class of error.